src/wind_energy/ABLWallFunction.cpp

```diff
@@ -37,14 +37,12 @@ void ABLWallFunction::init_log_law_height()
         // machine zero reg tests
         // eventually turn on pre_advance_work in InitialIterations() and delete this... or make a pre_timestep_work function?
         auto geom = m_mesh.Geom();
-        amrex::Real vx = 0.0;
-        amrex::Real vy = 0.0;
         amrex::ParmParse pp("incflo");
-        pp.query("ic_u", vx);
-        pp.query("ic_v", vy);
-        m_umean[0] = vx;
-        m_umean[1] = vy;
-        const amrex::Real uground = std::sqrt(vx * vx + vy * vy);
+        amrex::Vector<amrex::Real> vel{{0.0, 0.0, 0.0}};
+        pp.queryarr("velocity", vel);
+        m_umean[0] = vel[0];
+        m_umean[1] = vel[1];
+        const amrex::Real uground = std::sqrt(vel[0] * vel[0] + vel[1] * vel[1]);
```

Comment:
Ganesh is going to want to run with freestream in z-direction and terrain normal to y.